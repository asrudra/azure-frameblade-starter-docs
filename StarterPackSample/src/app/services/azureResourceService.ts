import { Subscription } from '../shared/models/subscription';

export interface GetSubscriptionParameters {
    apiVersion: string;
    authorizationToken: string;
}
export function getSubscriptions(getSubscriptionParameters: GetSubscriptionParameters): Promise<Subscription[]> {
    throw Error('not implemented');
}
