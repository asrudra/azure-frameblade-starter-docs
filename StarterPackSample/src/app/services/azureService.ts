import axios, { AxiosRequestConfig } from 'axios';
import { ContinuingResultSet } from './models/continuingResultSet';
import { AzureSubscription } from './models/azureSubscription';

export interface GetSubscriptionParameters {
    armEndpoint: string;
    authorizationToken: string;
    nextLink: string;
}
export function getAzureSubscriptions(parameters: GetSubscriptionParameters): Promise<ContinuingResultSet<AzureSubscription>> {
    const url: string = parameters.nextLink ? parameters.nextLink : `${parameters.armEndpoint}/subscriptions?api-version=2016-06-01`;
    const config: AxiosRequestConfig = {
        headers: {
            Authorization: `Bearer ${parameters.authorizationToken}`,
            ContentType: 'application/json'
        }
    };
    
    return axios.get(url, config)
        .then((response) => {
            return {
                items: response.data.value,
                nextLink: response.data.nextLink ? response.data.nextLink : '',
            };
        })
        .catch((error) => {
            throw Error(error);
        });
}
