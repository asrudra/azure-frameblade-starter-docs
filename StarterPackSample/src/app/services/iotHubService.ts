import { IotHubClient } from 'microsoft-azure-devices-iothub-controlplane20180401';
import { IotHubDescription } from '@iotHubControlPlane/lib/models';
import { getIotHubClientInstance } from './factories/iotHubClientFactory';
import { AzureResource } from './models/azureResource';

export interface GetIotHubDescriptionParameters {
    armEndpoint: string;
    authorizationToken: string;
    resource: AzureResource;
}
export function getIotHubDescription(parameters: GetIotHubDescriptionParameters): Promise<IotHubDescription> {
    
    const  client = getIotHubClientInstance({
        baseUri: parameters.armEndpoint,
        subscriptionId: parameters.resource.subscriptionId,
        token: parameters.authorizationToken
    });

    return client.iotHubResource.get(
        parameters.resource.resourceGroupName,
        parameters.resource.resourceName
    );
}

export interface UpdateIotHubDescriptionParameters {
    armEndpoint: string;
    authorizationToken: string;
    iotHubDescription: IotHubDescription;
    resource: AzureResource;
}
export function updateIotHubDescription(parameters: UpdateIotHubDescriptionParameters, iotHubClient?: IotHubClient): Promise<IotHubDescription> {

    const client = getIotHubClientInstance({
        baseUri: parameters.armEndpoint,
        subscriptionId: parameters.resource.subscriptionId,
        token: parameters.authorizationToken
    });
      
    return client.iotHubResource.createOrUpdate(
        parameters.resource.resourceGroupName,
        parameters.resource.resourceName,
        parameters.iotHubDescription
    );
}