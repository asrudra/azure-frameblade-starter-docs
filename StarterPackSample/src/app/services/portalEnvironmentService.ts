import { generateUuid } from 'ms-rest-js';
import { AzureResource } from './models/azureResource';
import { EnvironmentSettingNames } from './models/environmentSettingNames';

declare var frameSignature: string;
export let initialized: boolean = false;
export let parentOrigin: string;
export let promiseResolutions: Map<string, (value?: any | PromiseLike<any>) => void>; // tslint:disable-line: no-any
export let registeredActions: Map<string, (value: any) => void>; // tslint:disable-line: no-any    

export function registerActionOnChange(changeType: string, action: (value: any) => void): void {  // tslint:disable-line: no-any
    if (!initialized) {
        initialize();
    }
    
    registeredActions.set(changeType, action);  
}

export function getSetting(settingName: EnvironmentSettingNames): Promise<string> {
    return executeSettingRequest(settingName);
}

export function getAzureResource(): Promise<AzureResource> {
    return executeSettingRequest(EnvironmentSettingNames.AZURE_RESOURCE);
}

export function receiveMessage(event: MessageEvent) {
    if (event.origin.toLowerCase() !== parentOrigin) {
        return;
    }

    const msg = event.data;
    if (!msg || msg.signature !== frameSignature) {
        return;
    }

    const promise = promiseResolutions.get(msg.data.requestId);
    if (promise) {
        promise(msg.data.responseBody);
        promiseResolutions.delete(msg.data.requestId);
        return;
    }

    const registeredAction = registeredActions.get(msg.data.requestId);
    if (registeredAction) {
        registeredAction(msg.data.responseBody);
    }
}

function executeSettingRequest<T>(settingName: string): Promise<T> {
    if (!initialized) {
        initialize();
    }

    const requestUniqueId = generateUuid();
    const settingPromise = new Promise<T>((resolve, reject) => {
        window.parent.postMessage(
        {
            data: {
                requestId: requestUniqueId,
            },
            kind: settingName,
            signature: frameSignature
        }, 
        parentOrigin);
        promiseResolutions.set(requestUniqueId, resolve);
    });

    return settingPromise;
}

export function initialize(locationSearch?: string) {
    promiseResolutions = new Map<string, (value?: any | PromiseLike<any>) => void>();  // tslint:disable-line: no-any
    registeredActions = new Map<string, (value: any) => void>(); // tslint:disable-line: no-any
    
    if (!locationSearch) {
        locationSearch = window.location.search;
    }
    
    parentOrigin = getParentOrigin(locationSearch);
    if (isTrustedOrigin(parentOrigin)) {
        window.addEventListener('message', receiveMessage);
        window.parent.postMessage(
            {
                kind: 'ready',
                signature: frameSignature
            }, 
            parentOrigin); 
    }
    initialized = true;    
    
}

export function getParentOrigin(locationSearch: string): string {

    const queryStringMap = () => {
        const query = locationSearch.substring(1);
        const parameterList = query.split('&');
        const map: Map<string, string> = new Map();
        for (const item of parameterList) {
            const pair = item.split('=');
            map[decodeURIComponent(pair[0])] = decodeURIComponent(pair[1]);
        }
        return map;
    };
    
    const trustedAuthorityKey = 'trustedAuthority';
    return queryStringMap()[trustedAuthorityKey] || '';
}

export function isTrustedOrigin(originToVerify: string): boolean {
    
    const allowedParentOrigins: string[] = [
        'localhost', 
        'ms.portal.azure.com',
        'portal.azure.com',
        'portal.azure.us',
        'portal.azure.cn',
        'portal.microsoftazure.de',
        'df.onecloud.azure-test.net'
    ];

    const trustedAuthority = (originToVerify.split('//')[1] || '').toLowerCase();

    return allowedParentOrigins.some((origin: string) => {
        return origin === trustedAuthority;
    });
}