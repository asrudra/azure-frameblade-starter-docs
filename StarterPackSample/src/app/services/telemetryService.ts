import { AppInsights } from 'applicationinsights-js';

export function initialize() {
    if (AppInsights.downloadAndSetup) {
        AppInsights.downloadAndSetup({ instrumentationKey: 'enter key'});
    }
}

export interface LogEventParameters {
    eventName: string;
    measurements?:  { [name: string]: number };
    properties?:  { [name: string]: string };
}
export function logEvent(parameters: LogEventParameters) {
    AppInsights.trackEvent(
        parameters.eventName, 
        parameters.properties, 
        parameters.measurements);
}

export enum ExceptionSeverityLevel {
    Critical = 4,
    Error = 3,
    Information = 2,
    Verbose = 1,
    Warning = 0
}
export interface LogExceptionParameters {
    error: Error;
    handledAt: string;
    measurements?:  { [name: string]: number };
    properties?:  { [name: string]: string };
    severityLevel: ExceptionSeverityLevel;
}
export function logException(parameters: LogExceptionParameters) {
    AppInsights.trackException(
        parameters.error, 
        parameters.handledAt, 
        parameters.properties, 
        parameters.measurements, 
        parameters.severityLevel.valueOf());
}

export interface LogPageViewParameters {
    duration?: number;
    measurements?:  { [name: string]: number };
    properties?:  { [name: string]: string };
    url?: string;
    viewName: string;
    
}
export function logPageView(parameters: LogPageViewParameters) {
    AppInsights.trackPageView(
        parameters.viewName, 
        parameters.url, 
        parameters.properties, 
        parameters.measurements,
        parameters.duration);
}

export interface LogTraceParameters {
    message: string;
    properties?:  { [name: string]: string };
    severityLevel: ExceptionSeverityLevel;
}
export function logTrace(parameters: LogTraceParameters) {
    AppInsights.trackTrace(parameters.message, parameters.properties, parameters.severityLevel.valueOf()); 
}