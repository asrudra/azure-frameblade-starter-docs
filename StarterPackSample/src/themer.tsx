import * as React from 'react';
import { Fabric, Customizer, createTheme } from 'office-ui-fabric-react';
import { IPartialTheme } from 'office-ui-fabric-react/lib/Styling';
import { registerActionOnChange } from './app/services/portalEnvironmentService';
import { THEME_DARK, THEME_LIGHT, SCOPED_SETTINGS } from './app/constants';

export enum Theme {
    light = 'light',
    dark = 'dark',
}

interface ThemeState {
    theme: Theme; 
    officeTheme: IPartialTheme;
}

export class Themer extends React.Component<{}, ThemeState> {

    public constructor(props: {}) {
        super(props);
        this.state = {
            officeTheme: THEME_LIGHT,
            theme: Theme.light
        };

        this.registerThemeChange();
    }

    public render(): JSX.Element {

        const currentTheme = createTheme(this.state.officeTheme);

        return (
            <Customizer settings={{ theme: { ...currentTheme }}} scopedSettings={{...SCOPED_SETTINGS}}>
                <Fabric>
                    <div className={this.state.theme === Theme.dark ? 'theme-dark' : 'theme-light'}>
                        <div className="app">
                            {this.props.children}
                        </div>
                    </div>        
                </Fabric>  
            </Customizer>
        );
    }

    private registerThemeChange() {
        
        registerActionOnChange('themeChange', (result) => {
            const theme: IPartialTheme = result === Theme.dark ? THEME_DARK : THEME_LIGHT;
            this.setState({
                officeTheme: theme,
                theme: result
                
            });
        });
    }
}