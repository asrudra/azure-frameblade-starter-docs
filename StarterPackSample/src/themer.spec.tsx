import * as React from 'react';
import { shallow } from 'enzyme';
import { receiveMessage } from './app/services/portalEnvironmentService';
import { Theme, Themer } from './themer';

describe('themer', () => {
    context('initialization', () => {
        it ('matches snapshot', () => {
            const wrapper = shallow(<Themer/>);
            expect(wrapper).toMatchSnapshot();
        });
    
        it('initializes to light theme', () => {
            const wrapper = shallow(<Themer/>);
            expect(wrapper.state('theme')).toEqual(Theme.light);
        });
    });

    context('theme switching', () => {

        it ('switches to dark theme', () => {
            const messageEvent = {
                data: { 
                    data: {
                        requestId: 'themeChange',
                        responseBody: 'dark',
                    },
                    signature: 'portalEnvironmentFrameSignature'
                },
                origin: 'https://localhost'
            };
            const wrapper = shallow(<Themer/>);
            expect(wrapper.find('div.theme-light').exists()).toEqual(true);
            receiveMessage(messageEvent as MessageEvent);
    
            wrapper.update();
            expect(wrapper.find('div.theme-dark').exists()).toEqual(true);
            expect(wrapper.state('theme')).toEqual(Theme.dark);
        });

        it ('switches to light theme', () => {
            const messageEvent = {
                data: { 
                    data: {
                        requestId: 'themeChange',
                        responseBody: 'light',
                    },
                    signature: 'portalEnvironmentFrameSignature'
                },
                origin: 'https://localhost'
            };
            
            const wrapper = shallow(<Themer/>);
            wrapper.setState({
                theme: Theme.dark
            });
            
            wrapper.update();
            expect(wrapper.find('div.theme-dark').exists()).toEqual(true);
            receiveMessage(messageEvent as MessageEvent);
            
            wrapper.update();
            expect(wrapper.find('div.theme-light').exists()).toEqual(true);
            expect(wrapper.state('theme')).toEqual(Theme.light);
        });
    });
});