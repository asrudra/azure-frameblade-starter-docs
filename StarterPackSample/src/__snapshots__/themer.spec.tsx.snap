// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themer initialization matches snapshot 1`] = `
<Customizer
  scopedSettings={
    Object {
      "TextField": Object {
        "styles": Object {
          "root": Object {
            "fontSize": "12px",
            "selectors": Object {
              "input": Object {
                "fontSize": "12px",
              },
            },
          },
        },
      },
      "Toggle": Object {
        "styles": Object {
          "root": Object {
            "fontSize": "12px",
          },
        },
      },
    }
  }
  settings={
    Object {
      "theme": Object {
        "disableGlobalClassNames": false,
        "fonts": Object {
          "large": Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "fontFamily": "'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
            "fontSize": "17px",
            "fontWeight": 300,
          },
          "medium": Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
            "fontSize": "14px",
            "fontWeight": 400,
          },
          "mediumPlus": Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
            "fontSize": "15px",
            "fontWeight": 400,
          },
          "mega": Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
            "fontSize": "72px",
            "fontWeight": 100,
          },
          "small": Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
            "fontSize": "12px",
            "fontWeight": 400,
          },
          "smallPlus": Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
            "fontSize": "13px",
            "fontWeight": 400,
          },
          "superLarge": Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
            "fontSize": "42px",
            "fontWeight": 100,
          },
          "tiny": Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
            "fontSize": "10px",
            "fontWeight": 600,
          },
          "xLarge": Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
            "fontSize": "21px",
            "fontWeight": 100,
          },
          "xSmall": Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
            "fontSize": "11px",
            "fontWeight": 400,
          },
          "xxLarge": Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
            "fontSize": "28px",
            "fontWeight": 100,
          },
        },
        "isInverted": false,
        "palette": Object {
          "accent": "#0078d4",
          "black": "#1d1d1d",
          "blackTranslucent40": "rgba(0,0,0,.4)",
          "blue": "#0078d4",
          "blueDark": "#002050",
          "blueLight": "#00bcf2",
          "blueMid": "#00188f",
          "bodyBackground": "#fff",
          "bodyText": "#333",
          "green": "#107c10",
          "greenDark": "#004b1c",
          "greenLight": "#bad80a",
          "magenta": "#b4009e",
          "magentaDark": "#5c005c",
          "magentaLight": "#e3008c",
          "neutralDark": "#272727",
          "neutralLight": "#eaeaea",
          "neutralLighter": "#f4f4f4",
          "neutralLighterAlt": "#f8f8f8",
          "neutralPrimary": "#333",
          "neutralPrimaryAlt": "#4b4b4b",
          "neutralQuaternary": "#d0d0d0",
          "neutralQuaternaryAlt": "#dadada",
          "neutralSecondary": "#858585",
          "neutralSecondaryAlt": "#767676",
          "neutralTertiary": "#c2c2c2",
          "neutralTertiaryAlt": "#c8c8c8",
          "orange": "#d83b01",
          "orangeLight": "#ea4300",
          "orangeLighter": "#ff8c00",
          "purple": "#5c2d91",
          "purpleDark": "#32145a",
          "purpleLight": "#b4a0ff",
          "red": "#e81123",
          "redDark": "#a80000",
          "teal": "#008272",
          "tealDark": "#004b50",
          "tealLight": "#00b294",
          "themeDark": "#005a9e",
          "themeDarkAlt": "#106ebe",
          "themeDarker": "#004578",
          "themeLight": "#c7e0f4",
          "themeLighter": "#deecf9",
          "themeLighterAlt": "#eff6fc",
          "themePrimary": "#0078d4",
          "themeSecondary": "#2b88d8",
          "themeTertiary": "#71afe5",
          "white": "#fff",
          "whiteTranslucent40": "rgba(255,255,255,.4)",
          "yellow": "#ffb900",
          "yellowLight": "#fff100",
        },
        "semanticColors": Object {
          "actionLink": "#333",
          "actionLinkHovered": "#272727",
          "blockingBackground": "rgba(234, 67, 0, .2)",
          "bodyBackground": "#fff",
          "bodyDivider": "#eaeaea",
          "bodyFrameBackground": "#fff",
          "bodyFrameDivider": "#eaeaea",
          "bodyStandoutBackground": "#f8f8f8",
          "bodySubtext": "#858585",
          "bodyText": "#333",
          "bodyTextChecked": "#1d1d1d",
          "buttonBackground": "#f4f4f4",
          "buttonBackgroundChecked": "#c8c8c8",
          "buttonBackgroundCheckedHovered": "#eaeaea",
          "buttonBackgroundDisabled": "#f4f4f4",
          "buttonBackgroundHovered": "#eaeaea",
          "buttonBackgroundPressed": "#eaeaea",
          "buttonBorder": "transparent",
          "buttonBorderDisabled": "transparent",
          "buttonText": "#333",
          "buttonTextChecked": "#272727",
          "buttonTextCheckedHovered": "#1d1d1d",
          "buttonTextDisabled": "#c2c2c2",
          "buttonTextHovered": "#272727",
          "buttonTextPressed": "#272727",
          "defaultStateBackground": "#eaeaea",
          "disabledBackground": "#f4f4f4",
          "disabledBodyText": "#c8c8c8",
          "disabledSubtext": "#d0d0d0",
          "disabledText": "#c2c2c2",
          "errorBackground": "rgba(232, 17, 35, .2)",
          "errorText": "#a80000",
          "focusBorder": "#858585",
          "inputBackground": "#fff",
          "inputBackgroundChecked": "#0078d4",
          "inputBackgroundCheckedHovered": "#106ebe",
          "inputBorder": "#c2c2c2",
          "inputBorderHovered": "#272727",
          "inputFocusBorderAlt": "#0078d4",
          "inputForegroundChecked": "#fff",
          "inputPlaceholderText": "#858585",
          "link": "#0078d4",
          "linkHovered": "#004578",
          "listBackground": "#fff",
          "listHeaderBackgroundHovered": "#f4f4f4",
          "listHeaderBackgroundPressed": "#eaeaea",
          "listItemBackgroundChecked": "#eaeaea",
          "listItemBackgroundCheckedHovered": "#dadada",
          "listItemBackgroundHovered": "#f4f4f4",
          "listText": "#333",
          "listTextColor": "#333",
          "menuBackground": "#fff",
          "menuDivider": "#c8c8c8",
          "menuHeader": "#0078d4",
          "menuIcon": "#0078d4",
          "menuItemBackgroundChecked": "#eaeaea",
          "menuItemBackgroundHovered": "#f4f4f4",
          "menuItemBackgroundPressed": "#eaeaea",
          "menuItemText": "#333",
          "menuItemTextHovered": "#272727",
          "primaryButtonBackground": "#0078d4",
          "primaryButtonBackgroundDisabled": "#f4f4f4",
          "primaryButtonBackgroundHovered": "#106ebe",
          "primaryButtonBackgroundPressed": "#005a9e",
          "primaryButtonBorder": "transparent",
          "primaryButtonText": "#fff",
          "primaryButtonTextDisabled": "#d0d0d0",
          "primaryButtonTextHovered": "#fff",
          "primaryButtonTextPressed": "#fff",
          "smallInputBorder": "#858585",
          "successBackground": "rgba(186, 216, 10, .2)",
          "variantBorder": "#eaeaea",
          "variantBorderHovered": "#c2c2c2",
          "warningBackground": "rgba(255, 185, 0, .2)",
          "warningHighlight": "#ffb900",
          "warningText": "#333333",
        },
        "spacing": Object {
          "l1": "20px",
          "l2": "32px",
          "m": "16px",
          "s1": "8px",
          "s2": "4px",
        },
        "typography": Object {
          "families": Object {
            "default": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
            "monospace": "Menlo, Monaco, \\"Courier New\\", monospace",
          },
          "sizes": Object {
            "large": "17px",
            "medium": "14px",
            "mediumPlus": "15px",
            "mega": "72px",
            "mini": "10px",
            "small": "12px",
            "smallPlus": "13px",
            "xLarge": "21px",
            "xSmall": "11px",
            "xxLarge": "28px",
          },
          "variants": Object {
            "caption": Object {
              "color": undefined,
              "disabledColor": undefined,
              "family": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "hoverColor": undefined,
              "size": "11px",
              "weight": undefined,
            },
            "default": Object {
              "color": undefined,
              "disabledColor": undefined,
              "family": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "hoverColor": undefined,
              "size": "inherit",
              "weight": undefined,
            },
            "h1": Object {
              "color": undefined,
              "disabledColor": undefined,
              "family": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "hoverColor": undefined,
              "size": "72px",
              "weight": 100,
            },
            "h2": Object {
              "color": undefined,
              "disabledColor": undefined,
              "family": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "hoverColor": undefined,
              "size": "28px",
              "weight": 100,
            },
            "h3": Object {
              "color": undefined,
              "disabledColor": undefined,
              "family": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "hoverColor": undefined,
              "size": "21px",
              "weight": 100,
            },
            "h4": Object {
              "color": undefined,
              "disabledColor": undefined,
              "family": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "hoverColor": undefined,
              "size": "17px",
              "weight": 100,
            },
            "h5": Object {
              "color": undefined,
              "disabledColor": undefined,
              "family": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "hoverColor": undefined,
              "size": "15px",
              "weight": 100,
            },
            "link": Object {
              "color": "#0078d4",
              "disabledColor": undefined,
              "family": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "hoverColor": "#004578",
              "size": "inherit",
              "weight": undefined,
            },
          },
          "weights": Object {
            "bold": 700,
            "default": 400,
            "light": 100,
            "regular": 400,
            "semibold": 600,
          },
        },
      },
    }
  }
>
  <StyledFabricBase>
    <div
      className="theme-light"
    >
      <div
        className="app"
      />
    </div>
  </StyledFabricBase>
</Customizer>
`;
